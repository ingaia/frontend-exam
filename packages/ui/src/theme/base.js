import { css } from '@emotion/core';
import emotionNormalize from 'emotion-normalize';

const base = (theme) => css`
  @import url('https://fonts.googleapis.com/css2?family=Roboto+Condensed&display=swap');

  ${emotionNormalize};

  *,
  ::before,
  ::after {
    box-sizing: border-box;
  }

  html {
    font-family: ${theme.html.fontFamily};
    font-size: ${theme.html.fontSize};
    line-height: ${theme.html.lineHeight};
  }

  body {
    background-color: ${theme.body.bgColor};
    color: ${theme.body.fontColor};
  }
`;

export default base;
