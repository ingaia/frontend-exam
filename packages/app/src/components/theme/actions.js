export const setTheme = theme => ({
  type: 'SET_THEME',
  payload: theme,
});

export const setDecorate = decorate => ({
  type: 'SET_DECORATE',
  payload: decorate,
});

export default { setTheme, setDecorate };
