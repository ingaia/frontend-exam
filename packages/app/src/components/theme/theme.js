import React from 'react';
import { connect } from 'react-redux';
import { Theme } from '@witcher-page/ui';

const ReduxTheme = ({children, theme, ...props}) => (
  <Theme setGlobalStyles theme={theme} {...props}>
    {children}
  </Theme>
);

const mapState = state => ({ theme: state.theme });
const ConnectedTheme = connect(mapState)(ReduxTheme);

export default ConnectedTheme;
