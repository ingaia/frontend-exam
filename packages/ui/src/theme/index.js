import {
  useTheme as ut,
  withTheme as wt,
  Theme as T,
} from './theme';

export const useTheme = ut;
export const withTheme = wt;
export const Theme = T;

export default {
  useTheme,
  withTheme,
  Theme,
};
