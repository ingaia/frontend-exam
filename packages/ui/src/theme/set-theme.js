const fontFallback = 'system-ui, -apple-system, Segoe UI, Roboto, Ubuntu, Cantarell, Noto Sans, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji"';

const setTheme = (dark = true) => ({
  html: {
    fontFamily: `"Roboto Condensed", ${fontFallback}`,
    fontSize: '14px',
    lineHeight: '1.5',
  },
  body: {
    fontColor: dark ? '#fff' : '#1d1a1a',
    bgColor: dark ? '#1d1a1a' : '#fff',
  },
  container: {
    bgColor: dark ? '#040404' : '#fff',
  },
  masthead: {
    borderColor: dark ? '#4c4c4c' : '#a99e7e',
    bgColor: dark ? '#000' : '#fff',
  },
  input: {
    fontColor: dark ? '#fff' : '#1d1a1a',
    borderColor: '#a99e7e',
    focusBorderColor: dark ? '#fff' : '#1d1a1a',
    placeholderColor: '#a99e7e',
  },
  gradient: {
    start: dark ? '#000' : '#fff',
    end: dark ? 'rgba(0, 0, 0, 0)' : 'rgba(255, 255, 255, 0)',
  },
  scroll: {
    bgColor: dark ? '#1d1a1a' : '#fff',
    trackColor: '#a99e7e',
  },
  focus: {
    outlineColor: 'rgba(169, 158, 126, .5)',
  },
  button: {
    fontColor: dark ? '#fff' : '#1d1a1a',
    fontColorHover: '#a99e7e',
    bgColor: 'transparent',
    bgColorHover: 'transparent',
    activeFontColor: '#a99e7e',
    activeFontColorHover: '#fff',
    activeBgColor: 'transparent',
    activeBgColorHover: '#a99e7e',
    borderColor: '#a99e7e',
  },
  loader: {
    offColor: dark ? '#4c4c4c' : '#ccc',
    onColor: '#ff3509',
  },
  card: {
    borderColor: '#a99e7e',
    borderColorHover: dark ? '#fff' : '#1d1a1a',
  },
  modal: {
    borderColor: '#a99e7e',
    bgColor: dark ? 'rgba(29, 26, 26, 0.9)' : 'rgba(255, 255, 255, 0.9)',
  },
  assetsPath: dark ? 'assets/dark/' : 'assets/light/',
});

export default setTheme;
