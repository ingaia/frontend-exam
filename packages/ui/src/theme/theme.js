import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider, withTheme as wt, useTheme as ut } from 'emotion-theming';
import { Global } from '@emotion/core';
import setTheme from './set-theme';
import base from './base';

export const useTheme = ut;
export const withTheme = wt;

export const Theme = ({ theme, children, setGlobalStyles }) => {
  const selectedTheme = setTheme(theme === 'dark');
  const baseStyles = base(selectedTheme);

  return (
    <ThemeProvider theme={selectedTheme}>
      {setGlobalStyles && (<Global styles={baseStyles} />)}
      {children}
    </ThemeProvider>
  );
};

Theme.defaultProps = {
  theme: 'dark',
  setGlobalStyles: false,
};

Theme.propTypes = {
  theme: PropTypes.oneOf(['dark', 'light']),
  setGlobalStyles: PropTypes.bool,
  children: PropTypes.node.isRequired,
};

export default {
  withTheme,
  useTheme,
  Theme,
};
