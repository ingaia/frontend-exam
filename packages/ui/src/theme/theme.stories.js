import React from 'react';
import { Theme, useTheme } from './index';

export default { title: 'Theme' };

const TestComp = () => {
  const theme = useTheme();

  return (
    <div style={{ padding: '20px' }}>
      <p>
        Set the theme context as dark or light
      </p>
      <p>
        font:
        {theme.html.fontFamily}
      </p>
      <p>
        background color:
        {theme.body.bgColor}
      </p>
    </div>
  );
};

export const usingThemes = () => (
  <Theme setGlobalStyles theme="light">
    <TestComp />
  </Theme>
);
