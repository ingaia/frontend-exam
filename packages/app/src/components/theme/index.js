import ConnectedTheme from './theme';
import reducer from './reducer';
import actions from './actions';

export const Theme = ConnectedTheme;
export const themeReducer = reducer;
export const setTheme = actions.setTheme;
export const setDecorate = actions.setDecorate;

export default {
  Theme,
  reducer,
  setTheme,
  setDecorate,
};
