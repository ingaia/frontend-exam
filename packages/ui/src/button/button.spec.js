import React from 'react';
import renderer from 'react-test-renderer';
import serializer from 'jest-emotion';
import Button from './button';
import { Theme } from '../theme';

expect.addSnapshotSerializer(serializer);

test('renders button with correct styles', () => {
  const button = renderer.create(
    <Theme>
      <Button kind="active" href="#" link>Test</Button>
    </Theme>
  ).toJSON();

  expect(button).toMatchSnapshot();
});
