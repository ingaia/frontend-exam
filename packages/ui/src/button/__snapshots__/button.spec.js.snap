// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders button with correct styles 1`] = `
.emotion-0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  cursor: pointer;
  display: inline-block;
  position: relative;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  white-space: nowrap;
  background-color: transparent;
  border-width: 1px;
  border-style: solid;
  border-color: transparent;
  color: #fff;
  font-size: 0.857142857rem;
  height: 2.785714286rem;
  line-height: 39px;
  padding: 0 1em;
  text-transform: uppercase;
  -webkit-transition: color 0.5s ease;
  transition: color 0.5s ease;
  border-color: #a99e7e;
  color: #a99e7e;
  -webkit-transition: background-color 0.5s ease;
  transition: background-color 0.5s ease;
}

.emotion-0:hover,
.emotion-0:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0:focus {
  outline: 0;
  box-shadow: 0 0 0 2px rgba(169,158,126,.5);
}

@media screen and (min-width:780px) {
  .emotion-0 {
    font-size: 1rem;
  }
}

.emotion-0:hover {
  color: #a99e7e;
  border-color: #a99e7e;
}

.emotion-0:hover {
  color: #fff;
  background-color: #a99e7e;
}

<a
  className="emotion-0"
  href="#"
  role="button"
>
  Test
</a>
`;
